enginetest/memory_session.go

```diff
@@ -0,0 +1,219 @@
+// Copyright 2022 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package enginetest
+
+import (
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/sirupsen/logrus"
+)
+
+type InMemoryBaseSession struct {
+	Session sql.Session
+	idx     int
+}
```

Comment:
Is it possible to use InMemoryPersistedSession instead of creating/maintaining this new type?

Could we create a new constructor method, like NewInMemoryPersistedSessionWithValidationCallback that takes a function that ValidateSession could just call?

Other than that, I thought this test looked great!